### PR TITLE
Add support for cluster names with path separators

### PR DESCRIPTION
**Pull request by palemtnrider**
Closes: #1213  
* #1213 

**What changed?**
Instead of assuming the relative path between applications and apps, calculate it.

**Why?**
EKS clusters sometimes have a path separator in their name and this was causing an issue with Kustomize

**How did you test it?**
Added unit tests and existing tests will validate that existing functionality still works.

**Release notes**
Yes.

**Documentation Changes**
No

### CODE REVIEW

**Review by luizbafilho — APPROVED** [review]

File: pkg/services/app/add_test.go [hunk 1]
```diff
@@ -329,7 +329,7 @@ var _ = Describe("New directory structure", func() {
 		Expect(manifestsByPath[filepath.Join(git.WegoRoot, git.WegoAppDir, addParams.Name, "kustomization.yaml")]).ToNot(BeNil())
 		cname, err := kubeClient.GetClusterName(context.Background())
 		Expect(err).To(BeNil())
-		clusterKustomizeFile := manifestsByPath[filepath.Join(git.WegoRoot, git.WegoClusterDir, cname, "/user/kustomization.yaml")]
+		clusterKustomizeFile := manifestsByPath[filepath.Join(git.WegoRoot, git.WegoClusterDir, cname, git.WegoClusterUserWorloadDir, "kustomization.yaml")]
```
Comment (yiannistri) [inline pkg/services/app/add_test.go:332]: s/WegoClusterUserWorloadDir/WegoClusterUserWorkloadDir

File: pkg/services/app/add_test.go [hunk 2]
```diff
@@ -359,5 +359,32 @@ var _ = Describe("New directory structure", func() {
 		Expect(r[0].(string)).To(Equal("../../../apps/" + origName))
 		Expect(r[1].(string)).To(Equal("../../../apps/" + addParams.Name))
 	})
+	It("deals with a cluster dir with additional subdirectories", func() {
```
Comment (yiannistri) [inline pkg/services/app/add_test.go:362]: Shall we add a line break between the `It` assertions? Unrelated to these changes but I noticed that this file is not consistent in terms of the line breaks.

File: pkg/services/gitopswriter/gitopswriter.go [hunk 3]
```diff
@@ -58,7 +58,12 @@ func (dw *gitOpsDirectoryWriterSvc) AddApplication(ctx context.Context, app mode
 
 	defer remover()
 
-	kManifest, err := addKustomizeResources(app, repoDir, clusterName, appKustomizeReference(app))
+	resourceEntry, err := appKustomizeReference(getUserKustomizationRepoPath(clusterName), appPath(app.Name))
+	if err != nil {
+		return err
```
Comment (yiannistri) [inline pkg/services/gitopswriter/gitopswriter.go:63]: Shall we add more context here before returning the %w error?

**Review by yiannistri — APPROVED** [review]
Some minor comments but overall this looks good 👍

File: pkg/services/app/add_test.go (hunk 1 above)
Comment (palemtnrider) [reply pkg/services/app/add_test.go:332]: 👍 thanks for catching this.  I saw it at one point but then forgot to fix it.

**Review by josecordaz — APPROVED** [review]
LGTM

File: pkg/services/app/add_test.go [hunk 4]
```diff
@@ -360,4 +361,32 @@ var _ = Describe("New directory structure", func() {
...
+		}
+
+		Expect(manifestsByPath[filepath.Join(git.WegoRoot, git.WegoAppDir, addParams.Name, "kustomization.yaml")]).ToNot(BeNil())
+		cname, err := kubeClient.GetClusterName(context.Background())
+		Expect(err).To(BeNil())
+		clusterKustomizeFile := manifestsByPath[filepath.Join(git.WegoRoot, git.WegoClusterDir, cname, git.WegoClusterUserWorkloadDir, "kustomization.yaml")]
+		Expect(clusterKustomizeFile).ToNot(BeNil())
+		manifestMap := map[string]interface{}{}
+
+		Expect(yaml.Unmarshal(clusterKustomizeFile, &manifestMap)).Should(Succeed())
```
[the remark this answers is not in the document]
Comment (palemtnrider) [reply pkg/services/app/add_test.go:382]: I prefer the shorter names, however, if you feel it makes the code easier to read 👍

**Pull request merged**